Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Db.ts

```diff
@@ -0,0 +1,111 @@
+  database: 'meadowlark',
+};
+
+/**
+ * Checks that the meadowlark schema, document and references tables exist in the database, if not will create them
+ * @param client The Postgres client for querying
+ */
+export async function checkExistsAndCreateTables(client: Client) {
+  try {
+    await client.query(createSchemaSql);
+    await client.query(createDocumentTableSql);
+    await client.query(createReferencesTableSql);
+  } catch (e) {
+    const message = e.constructor.name.includes('Error') ? e.message : 'unknown';
+    Logger.error(`Error connecting PostgreSql. Error was ${message}`, null);
+    throw e;
+  }
+}
+
+/**
+ * Create a connection pool, check that the database and table structure is in place
+ * or create it if not and then return the pool
+ */
+export async function createConnectionPoolAndReturnClient(): Promise<Client> {
+  let client: Client;
+  try {
+    // Attempt to connect to the meadowlark DB. If the meadowlark database doesn't exist, the connection will fail and throw
+    // an error. If this happens, we will create a client connection to the postgres database, create the meadowlark
+    // database, and disconnect. From there reconnect the pool to the meadowlark database and continue
+
+    dbPool = new Pool(dbConfiguration);
+    client = await dbPool.connect();
+
+    Logger.info(`Connected to ${dbConfiguration.database} successfully`, null);
+
+    return client;
+  } catch (e) {
+    const message = e.constructor.name.includes('Error') ? e.message : 'unknown';
+
+    // if this anything other than a DB doesn't exist error, there's a bigger problem and we don't want to continue
+    if (e.message !== `database "${dbConfiguration.database}" does not exist`) {
+      Logger.error(`Error connecting Postgres. Error was ${message}`, null);
+      throw e;
+    }
+    dbPool.end();
+  }
+
+  // The meadowlark DB doesn't exist, create a separate client that connects to the postgres(default) DB to create
+  // meadowlark DB, then reconnect the pool to the meadowlark DB and return
+  try {
+    const meadowlarkDbName = dbConfiguration.database;
+    dbConfiguration.database = 'postgres';
+
+    client = new Client(dbConfiguration);
+    client.connect();
+    await client.query(await GetCreateDatabaseSql(meadowlarkDbName));
+
+    Logger.info(`Database ${meadowlarkDbName} created successfully`, null);
+  } catch (e) {
+    const message = e.constructor.name.includes('Error') ? e.message : 'unknown';
+    Logger.error(`Error connecting Postgres. Error was ${message}`, null);
+    throw e;
+  } finally {
+    await client.end();
+  }
+
+  dbConfiguration.database = 'meadowlark';
+  dbPool = new Pool(dbConfiguration);
+
+  return dbPool.connect();
+}
+
+/**
+ * Return the shared client
+ */
+export async function getSharedClient(): Promise<Client> {
+  if (dbPool == null) {
+    const client = await createConnectionPoolAndReturnClient();
+    await checkExistsAndCreateTables(client);
```

Comment:
@stephenfuqua what about moving the checkExistsAndCreateTables call into the end of the createConnectionPoolAndReturnClient call? This would mean it would only check/create tables if the meadowlark database doesn't exist rather than every startup, leaving the happy path as 

1. Create Pool
2. Connect to Pool
3. Return connected client

Reply:
I thought about that too, but then what happens if someone pre-creates the empty database?

Hence let's just leave as is and defer until later, after we get feedback from the community about deployment approaches.